ikarus/finiteelements/mechanics/loads/traction.hh

```diff
@@ -0,0 +1,156 @@
+        neumannBoundaryLoad = p_neumannBoundaryLoad;
+
+      assert(((not p_neumannBoundary and not neumannBoundaryLoad) or (p_neumannBoundary and neumannBoundaryLoad))
+             && "If you pass a Neumann boundary you should also pass the function for the Neumann load!");
+    }
+
+    /**
+     * @brief Const accessor to the underlying displacement-based finite element (CRTP).
+     *
+     * @return Const reference to the underlying displacement-based finite element.
+     */
+    constexpr const DisplacementBasedElement& dbElement() const {
+      return static_cast<DisplacementBasedElement const&>(*this);
+    }
+
+    /**
+     * @brief Calculates the scalar energy for the given traction load.
+     *
+     * @tparam ScalarType The scalar type for the energy.
+     *
+     * @param par The FERequirementType object.
+     * @param dx Optional displacement vector.
+     * @return The scalar energy.
+     */
+    template <typename ScalarType>
+    auto calculateScalar(const FERequirementType& par,
+                         const std::optional<const Eigen::VectorX<ScalarType>>& dx = std::nullopt) const -> ScalarType {
+      const auto uFunction = dbElement().displacementFunction(par, dx);
+      ScalarType energy    = 0.0;
+      const auto& lambda   = par.getParameter(Ikarus::FEParameter::loadfactor);
+      auto element         = dbElement().localView().element();
+      if (not neumannBoundary and not neumannBoundaryLoad) return energy;
+
+      for (auto&& intersection : intersections(neumannBoundary->gridView(), element)) {
+        if (not neumannBoundary->contains(intersection)) continue;
+
+        const auto& quadLine = Dune::QuadratureRules<double, myDim - 1>::rule(intersection.type(), dbElement().order());
+
+        for (const auto& curQuad : quadLine) {
+          // Local position of the quadrature point
+          const Dune::FieldVector<double, myDim>& quadPos = intersection.geometryInInside().global(curQuad.position());
+
+          const double integrationElement = intersection.geometry().integrationElement(curQuad.position());
+
+          // The value of the local function
+          const auto uVal = uFunction.evaluate(quadPos);
+
+          // Value of the Neumann data at the current position
+          auto neumannValue = neumannBoundaryLoad(intersection.geometry().global(curQuad.position()), lambda);
+
+          energy -= neumannValue.dot(uVal) * curQuad.weight() * integrationElement;
+        }
+      }
+      return energy;
+    }
+
+    /**
+     * @brief Calculates the force vector for the given traction load.
+     *
+     * @tparam ScalarType The scalar type for the force vector.
+     *
+     * @param par The FERequirementType object.
+     * @param force The force vector to be updated.
+     * @param dx Optional displacement vector.
+     */
+    template <typename ScalarType>
+    void calculateVector(const FERequirementType& par, typename Traits::template VectorType<ScalarType> force,
```

Comment:
These function should follow the same style and in the other finite elements. Public non-templated methods calculateScalar, calculateVector, ... and protected: template memberfunctions calculateScalarImpl, calculateVectorImpl, ... with the actual implementation.